### DISCUSSION TITLE
After the parent note adds the #widget tag, the child note will also be executed automatically. Is this a feature or a bug?

### DISCUSSION THREAD

**Discussion by SiriusXT** [TriliumNext/Trilium#3778] [General]
I encountered a bug when debugging the code today, and I couldn’t find the error, and it didn’t work after commenting. Later, I found that the backup subnote was also running.

**Comment by zadam** [comment]
This is actually a feature. The tree structure also presents a dependency tree between different script notes.

Children JS notes are treated as CommonJS modules and exported into the parent notes. That means they are also executed and they are expected to expose something in the `module.exports = ...`.

Some of the scripting examples in the demo document are using this "feature".

**Comment by Nriver** [comment]
I'm using this feature to seperate the [code](https://github.com/Nriver/trilium-translation/blob/main/demo-cn/%E7%A4%BA%E4%BE%8B%E7%AC%94%E8%AE%B0%20-%20%E8%AF%B7%E4%B8%8D%E8%A6%81%E5%88%A0%E9%99%A4/Trilium%20%E6%89%A9%E5%B1%95/Trilium%20%E8%84%9A%E6%9C%AC%20script/startup%20%E5%90%AF%E5%8A%A8%E9%A1%B9/startup%20message%20%E5%90%AF%E5%8A%A8%E4%BF%A1%E6%81%AF.js) and [configuration](https://github.com/Nriver/trilium-translation/blob/main/demo-cn/%E7%A4%BA%E4%BE%8B%E7%AC%94%E8%AE%B0%20-%20%E8%AF%B7%E4%B8%8D%E8%A6%81%E5%88%A0%E9%99%A4/Trilium%20%E6%89%A9%E5%B1%95/Trilium%20%E8%84%9A%E6%9C%AC%20script/startup%20%E5%90%AF%E5%8A%A8%E9%A1%B9/startup%20message%20%E5%90%AF%E5%8A%A8%E4%BF%A1%E6%81%AF/config.js) in the simple demo.

I think it could be used for update of plugins without lost the old configuration. Could be benefit for plugin market in the future, if any.
